src/System.Diagnostics.DiagnosticSource/src/System/Diagnostics/Activity.cs

```diff
@@ -392,13 +404,14 @@ private string GenerateRootId()
                 Interlocked.CompareExchange(ref s_uniqPrefix, GenerateInstancePrefix(), null);
             }
 #if DEBUG
-            string ret = s_uniqPrefix + "-" + OperationName + "-" + Interlocked.Increment(ref s_currentRootId).ToString("x") + '.';
+            string ret = s_uniqPrefix + "-" + OperationName.Replace('.', '-') + "-" + Interlocked.Increment(ref s_currentRootId).ToString("x") + '.';
 #else       // To keep things short, we drop the operation name 
             string ret = s_uniqPrefix + "-" + Interlocked.Increment(ref s_currentRootId).ToString("x") + '.';
 #endif
             return ret;
         }
 
+        [SecuritySafeCritical]
```

Comment:
Actually the project just had broken dependencies. I've fixed it up, please see: https://github.com/ericstj/corefx/commit/0556f0246ef9709cbd6ee743f6b403f528d88881

To test this locally you can run `build -allConfigurations`.

After doing so once you can run msbuild /t:BuildAllConfigurations on any project you want.